docs/graph_engine/overview.rst

```diff
@@ -0,0 +1,194 @@
+Graph engine overview
+=====================
+
+Graphs for GNN can be stored in different formats and in order to have ability to plug them in the existing
+algorithms we define an API for basic operations in
+`_base.py <https://github.com/microsoft/DeepGNN/blob/main/src/python/deepgnn/graph_engine/_base.py#L30>`_.
+Typically nodes are represented as integers while edges are triple of integers:
+(source node, destination node, edge type). Each node and edge can store multiple features that can be addressed
+by an integer id. We support float/integer/binary feature types.
+
+
+Graph as dataset
+----------------
+
+Typically we want to traverse all nodes or edges in a graph to train some model. A straightforward way to do it
+in Pytorch is to subclass `IterableDataset <https://pytorch.org/docs/stable/data.html#iterable-style-datasets>`_
+and provide a node/edge iterator.
```

Comment:
Make clear we already include a subclass of IterableDataset, TorchDeepGNNDataset and how it can be used // point to example.

Reply:
I just convert the existing docs to rST.